src/main.ts

```diff
@@ -118,7 +118,10 @@ export const processIntent = async (intent: Intent) => {
     owner,
   });
 
-  for (const sourceChain of intent.sourceChains) {
+  const sourceChains = intent.sourceChains.length > 0 ? intent.sourceChains : fundingSupportedChains
+  const sourceTokens = intent.sourceTokens.length > 0 ? intent.sourceTokens : fundingSupportedTokens
```

Comment:
can we move this into fundAccount - specifically only inside the flag that enables funding

Reply:
fyi this tool is also used by external parties (mostly relayers) to do testing so the core flow should be geared to prod envs instead of the e2e flow